controller/scheduler/scheduler.go

```diff
@@ -278,6 +288,34 @@ func (s *Scheduler) streamHostEvents() error {
 	}
 }
 
+func (s *Scheduler) streamRouterEvents() {
+	log := s.logger.New("fn", "streamRouterEvents")
+
+	var events chan *discoverd.Event
+	var stream stream.Stream
```

Comment:
it this gets assigned, or if `connect` is called again (below) should the stream be closed?